### ISSUE TITLE
Implement a PreNorm layer

### ISSUE THREAD

**Issue by Rishit-dagli** [Rishit-dagli/Perceiver#1]
Create a Normalization layer from the `tf.keras.layerr.Layers`. This should essentially figure out the right axis and implement layer normalization on it.
